aptos-move/aptos-gas/src/algebra.rs

```diff
@@ -0,0 +1,226 @@
+use serde::{Deserialize, Serialize};
+use std::cmp::{Eq, Ord, Ordering, PartialEq, PartialOrd};
+use std::convert::From;
+use std::fmt::{self, Debug, Display};
+use std::marker::PhantomData;
+use std::ops::{Add, AddAssign, Mul, Sub, SubAssign};
+
+/***************************************************************************************************
+ * Units of Measurement
+ *
+ **************************************************************************************************/
+/// Unit of (external) gas.
+pub enum GasUnit {}
```

Comment:
That's an interesting idiom: a type which has no instance but can be used as a phantom type.